File: files/2025-11-02/CSEC 02112025.R
library(tidyverse)
library(readxl)
library(glue)
library(janitor)

path = "files/2025-11-02/Challenge 73.xlsx"
input = read_excel(path, range = "B3:F9")
test  = read_excel(path, range = "H3:I6")
test$Experience = str_replace(test$Experience, "-Twitter", "- Twitter")

result = input %>%
  clean_names() %>%
  mutate(rn = row_number(), .by = candidate) %>%
  transmute(
    Candidate = candidate,
    Experience = glue("{rn}. {from_date}:{to_date} - {past_position} - {past_employer}")) %>%
  summarise(Experience = str_c(Experience, collapse = "\r\n"), .by = Candidate)

all.equal(result,test) 
# [1] TRUE

File: files/2025-11-02/CSEC 02112025.py
import pandas as pd

path = "files/2025-11-02/Challenge 73.xlsx"
input = pd.read_excel(path, usecols="B:F", skiprows=2, nrows=7)
test = pd.read_excel(path, usecols="H:I", skiprows=2, nrows=3).rename(columns=lambda col: col.replace('.1', ''))
test['Experience'] = test['Experience'].str.replace('-Twitter', '- Twitter', regex=False)

input.columns = [c.strip().lower().replace(' ', '_') for c in input.columns]
input['rn'] = input.groupby('candidate').cumcount() + 1
input['Experience'] = input.apply(
    lambda r: f"{r['rn']}. {r['from_date']}:{r['to_date']} - {r['past__position']} - {r['past__employer']}", axis=1
)
result = input.groupby('candidate')['Experience'].apply('\n'.join).reset_index()
result.columns = ['Candidate', 'Experience']

print(result.equals(test)) # True